### PR TITLE
Fix build: get_current_dir_name() is a GNU extension

### PR DESCRIPTION
**Pull request by thiagomacieira**
So add a replacemenet function.

Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>

### CODE REVIEW

File: src/lib/common/sol-missing.h [hunk 1]
```diff
@@ -70,6 +70,39 @@ memmem(const void *haystack, size_t haystacklen, const void *needle, size_t need
...
+#include <unistd.h>
+
+static inline char *get_current_dir_name(void)
+{
+    size_t len = 0;
+    char *buffer = NULL;
+
+    while (1) {
+        len += PATH_MAX;
+        char *nb = (char *)realloc(buffer, len);
```
Comment (barbieri) [inline src/lib/common/sol-missing.h]: consider using our `sol-buffer.h`, much easier and handles realloc checks for you. Ask it for the size you want.

File: src/lib/common/sol-missing.h [hunk 2]
```diff
@@ -70,6 +70,39 @@ memmem(const void *haystack, size_t haystacklen, const void *needle, size_t need
 }
 #endif
 
+#if !defined(HAVE_GET_CURRENT_DIR_NAME) && defined(HAVE_UNIX)
```
Comment (thiagomacieira) [inline src/lib/common/sol-missing.h]: Note this depends on HAVE_UNIX being defined, which is in another patch not yet ready for submission. But this won't break the build.

File: src/lib/common/sol-missing.h (hunk 1 above)
Comment (thiagomacieira) [inline src/lib/common/sol-missing.h]: NAK. This function needs to return a pointer freeable with free().

Comment (barbieri) [inline src/lib/common/sol-missing.h]: you can keep the `struct sol_buffer` on the stack and use `sol_buffer_steal()` :-)

realloc is a PITA

Comment (thiagomacieira) [inline src/lib/common/sol-missing.h]: Not after you've already done it correctly :-)

The problem with doing with sol-buffer.h is that I'd have to learn it first and I don't have time for that.

Comment (barbieri) [inline src/lib/common/sol-missing.h]: Fair, go for it. +1

**Comment by bdilly** [comment]
OK, I'm merging this patch so the next guy working on this task may provide HAVE_UNIX definition.

Thank you

**Pull request closed without merging**